src/main/java/com/PhotomosaicSara/image/PhotomosaicService.java

```diff
@@ -0,0 +1,98 @@
+package com.PhotomosaicSara.image;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+import static com.PhotomosaicSara.image.Pixel.*;
+
+public class PhotomosaicService {
+
+    private static final int TILE_WIDTH = 90;
+    private static final int TILE_HEIGHT = 90;
+    private static final int TILE_SCALE = 9;
+
+    public BufferedImage makeOutputImage(int width, int height, Collection<BufferedImagePart> parts) {
+        final var image = new BufferedImage(width * TILE_SCALE, height * TILE_SCALE, BufferedImage.TYPE_3BYTE_BGR);
+
+        for (final var part : parts) {
+            final var imagePart = image.getSubimage(part.x * TILE_SCALE, part.y * TILE_SCALE, TILE_WIDTH, TILE_HEIGHT);
+            imagePart.setData(part.image.getData());
+        }
+        return image;
+    }
+
+    public Tile getBestFitTile(BufferedImage target, Collection<Tile> tiles) {
+        Tile bestFit = null;
+        int bestFitScore = -1;
+
+        for (final var tile : tiles) {
+            final var score = getScore(target, tile);
+            if (score > bestFitScore) {
+                bestFitScore = score;
+                bestFit = tile;
+            }
+        }
+        return bestFit;
+    }
+
+    private int getScore(BufferedImage target, Tile tile) {
+        assert target.getHeight() == Tile.scaledHeight;
```

Comment:
assert shouldn't be used at all. All valitaions should be done using if and exception mechanisms